app/models/censor_rule.rb

```diff
@@ -51,7 +51,14 @@ def apply_to_text(text_to_censor)
 
   def apply_to_binary(binary_to_censor)
     return nil if binary_to_censor.nil?
-    binary_to_censor.gsub(to_replace('ASCII-8BIT')) { |match| match.gsub(single_char_regexp, 'x') }
+
+    current_encoding = binary_to_censor.encoding
+    binary = binary_to_censor.dup
+    binary = binary.force_encoding('ASCII-8BIT')
+    binary.gsub!(to_replace('ASCII-8BIT')) do |match|
+      match.gsub(single_char_regexp, 'x')
+    end
+    binary.force_encoding(current_encoding)
```

Comment:
Rather than forcing the encoding of the binary, why not match the censor rule encoding to the binary at hand?:

```diff
diff --git a/app/models/censor_rule.rb b/app/models/censor_rule.rb
index 6ed0a71b6..94e7c2f82 100644
--- a/app/models/censor_rule.rb
+++ b/app/models/censor_rule.rb
@@ -51,7 +51,10 @@ class CensorRule < ApplicationRecord

   def apply_to_binary(binary_to_censor)
     return nil if binary_to_censor.nil?
-    binary_to_censor.gsub(to_replace('ASCII-8BIT')) { |match| match.gsub(single_char_regexp, 'x') }
+
+    binary_to_censor.gsub(to_replace(binary_to_censor.encoding)) do |match|
+      match.gsub(single_char_regexp, 'x')
+    end
   end
```

I _think_ this is better for a couple of reasons:

1. Its a bit of a cleaner change
2. If we have a UTF8 binary, we may end up losing accented characters as we force to ASCII and then back to UTF8? In this way, we convert any accents in `CensorRule#text`, but that's fine as we know they won't exist in the ASCII version anyway.

Either way seems to fix the problem 👍